pkg/kube/client_test.go

```diff
@@ -114,210 +118,209 @@ func newTestClient(t *testing.T) *Client {
+type RequestResponseLogClient struct {
+	t           *testing.T
+	cb          RoundTripperTestFunc
+	actionsLock sync.Mutex
+	Actions     []RequestResponseAction
+}
 
-		result, err := c.Create(list)
-		if err != nil {
-			t.Fatal(err)
-		}
+func (r *RequestResponseLogClient) Do(req *http.Request) (*http.Response, error) {
+	t := r.t
+	t.Helper()
 
-		if len(result.Created) != 1 {
-			t.Errorf("expected 1 resource created, got %d", len(result.Created))
+	readBodyBytes := func(body io.ReadCloser) []byte {
+		if body == nil {
+			return []byte{}
 		}
 
-		expectedActions := []string{
-			"/namespaces/default/pods:POST",
-			"/namespaces/default/pods:POST",
-			"/namespaces/default/pods:POST",
-		}
-		if len(expectedActions) != len(actions) {
-			t.Fatalf("unexpected number of requests, expected %d, got %d", len(expectedActions), len(actions))
-		}
-		for k, v := range expectedActions {
-			if actions[k] != v {
-				t.Errorf("expected %s request got %s", v, actions[k])
-			}
-		}
-	})
+		defer body.Close()
+		bodyBytes, err := io.ReadAll(body)
+		require.NoError(t, err)
 
-	t.Run("Create failure", func(t *testing.T) {
-		list, err := c.Build(objBody(&listB), false)
-		if err != nil {
-			t.Fatal(err)
-		}
+		return bodyBytes
+	}
 
-		_, err = c.Create(list)
-		if err == nil {
-			t.Errorf("expected error")
-		}
+	reqBytes := readBodyBytes(req.Body)
 
-		expectedString := "Operation cannot be fulfilled on resourcequotas \"quota\": the object has been modified; " +
-			"please apply your changes to the latest version and try again"
-		if !strings.Contains(err.Error(), expectedString) {
-			t.Errorf("Unexpected error message: %q", err)
-		}
+	t.Logf("Request: %s %s %s", req.Method, req.URL.String(), reqBytes)
+	if req.Body != nil {
+		req.Body = io.NopCloser(bytes.NewReader(reqBytes))
+	}
 
-		expectedActions := []string{
-			"/namespaces/default/pods:POST",
-		}
-		for k, v := range actions {
-			if expectedActions[0] != v {
-				t.Errorf("expected %s request got %s", v, actions[k])
-			}
-		}
+	resp, err := r.cb(r.Actions, req)
+
+	respBytes := readBodyBytes(resp.Body)
+	t.Logf("Response: %d %s", resp.StatusCode, string(respBytes))
+	if resp.Body != nil {
+		resp.Body = io.NopCloser(bytes.NewReader(respBytes))
+	}
+
+	r.actionsLock.Lock()
+	defer r.actionsLock.Unlock()
+	r.Actions = append(r.Actions, RequestResponseAction{
+		Request:  *req,
+		Response: *resp,
+		Error:    err,
 	})
+
+	return resp, err
 }
 
-func testUpdate(t *testing.T, threeWayMerge bool) {
-	t.Helper()
-	listA := newPodList("starfish", "otter", "squid")
-	listB := newPodList("starfish", "otter", "dolphin")
-	listC := newPodList("starfish", "otter", "dolphin")
-	listB.Items[0].Spec.Containers[0].Ports = []v1.ContainerPort{{Name: "https", ContainerPort: 443}}
-	listC.Items[0].Spec.Containers[0].Ports = []v1.ContainerPort{{Name: "https", ContainerPort: 443}}
+func TestCreate(t *testing.T) {
```

Comment:
`TestCreate` was moved from https://github.com/helm/helm/pull/31030/files#diff-c2e1286886bf105142600453bd088a883e6c667ee65eb98f9706ed3e4648707cR193 (diff is not great here). Original test cases remain the same.